spec/services/batch_reader_spec.rb

```diff
@@ -2,7 +2,24 @@
 require 'rails_helper'
 
 describe Hyrax::BatchIngest::BatchReader do
-  it 'passes tests' do
-    true.should == true
+  let(:reader) { described_class.new(source_location) }
+  let(:source_location) { 'path/to/batch/source' }
+
+  describe '#initialize' do
+    it 'stores the source_location' do
+      expect(reader.source_location).to eq source_location
+    end
+  end
+
+  describe '#submitter_email' do
+    it 'throws an exception' do
+      expect { reader.submitter_email }.to raise_error(Hyrax::BatchIngest::ReaderError)
+    end
+  end
+
+  describe '#batch_items' do
+    it 'throws an exception' do
+      expect { reader.batch_items }.to raise_error(Hyrax::BatchIngest::ReaderError)
+    end
```

Comment:
I thought the strategy was for implementers to overwrite `#read` and within their extended class's `#read` method they would then populate `@batch_items`, `@submitter_email`, etc. If that's the case, then they won't be overwriting the `#batch_items` or `#submitter_email` accessor methods, so these wouldn't raise errors.

Reply:
This is just the spec test for the abstract class reader.  The shared specs that I'm writing now will handle this better for actual reader implementations.  I'm also planning on rewriting this spec to use the shared specs.